stomp_plugins/src/cost_functions/cartesian_distance.cpp

```diff
@@ -0,0 +1,182 @@
+#include <math.h>
+#include <stomp_plugins/cost_functions/cartesian_distance.h>
+#include <XmlRpcException.h>
+#include <pluginlib/class_list_macros.h>
+#include <ros/console.h>
+
+PLUGINLIB_EXPORT_CLASS(stomp_moveit::cost_functions::CartesianDistance,stomp_moveit::cost_functions::StompCostFunction);
+
+namespace stomp_moveit
+{
+namespace cost_functions
+{
+
+CartesianDistance::CartesianDistance():
+    name_("CartesianDistance")
+{
+  // TODO Auto-generated constructor stub
+
+}
+
+CartesianDistance::~CartesianDistance()
+{
+  // TODO Auto-generated destructor stub
+}
+
+bool CartesianDistance::initialize(moveit::core::RobotModelConstPtr robot_model_ptr,
+                        const std::string& group_name,XmlRpc::XmlRpcValue& config)
+{
+  group_name_ = group_name;
+  robot_model_ = robot_model_ptr;
+
+  return configure(config);
+}
+
+bool CartesianDistance::configure(const XmlRpc::XmlRpcValue& config)
+{
+  using namespace XmlRpc;
+
+  try
+  {
+    XmlRpcValue params = config;
+
+    position_cost_weight_ = static_cast<double>(params["position_cost_weight"]);
+    orientation_cost_weight_ = static_cast<double>(params["orientation_cost_weight"]);
+    translation_tolerance_ =  static_cast<double>(params["translation_tolerance"]);
+    rotation_tolerance_ =  static_cast<double>(params["rotation_tolerance"]);
```

Comment:
Just a question: What will happen when the param is not in config? Does the operator[] throw?  Based on catch it look so however usual behavior of operator [] is undefined if you are accessing the non existing member.

Reply:
No this does not throw. I add check if parameter exists